service/controller/clusterapi/v29/adapter/guest_route_tables.go

```diff
@@ -17,17 +17,19 @@ type GuestRouteTablesAdapter struct {
 }
 
 func (r *GuestRouteTablesAdapter) Adapt(cfg Config) error {
+	workerAZs := key.SortedWorkerAvailabilityZones(cfg.MachineDeployment)
+
 	r.HostClusterCIDR = cfg.ControlPlaneVPCCidr
 	r.PublicRouteTableName = RouteTableName{
 		ResourceName: "PublicRouteTable",
-		TagName:      key.RouteTableName(cfg.CustomObject, suffixPublic, 0),
+		TagName:      key.RouteTableName(cfg.CustomObject, suffixPublic, workerAZs[0]),
```

Comment:
These magical "first AZ" entries made me worry a bit, but we need to tackle these together with overhaul for keeping stable master AZ as well.